packages/algob/src/internal/deployer.ts

```diff
@@ -68,6 +69,28 @@ class DeployerBasicMode {
     return found;
   }
 
+  /**
+   * Asserts if asset is present in checkpoint
+   * @param name Asset name
+   */
+  assertNoAsset (name: string): void {
+    if (this.isDefined(name)) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+      throw new BuilderError(
+        ERRORS.BUILTIN_TASKS.DEPLOYER_ASSET_ALREADY_PRESENT, {
+          assetName: name
+        });
+    }
+  }
```

Comment:
Could you please explain why we moved this function from *Deploy Mode* to *Basic Mode* ?

Reply:
so that we can use it both modes.
user can also check if checkpoint has particular asset in run mode